src/Tests/CaptainHook.Tests/Director/DirectorServiceTests.cs

```
@@ -221,5 +225,30 @@ public async Task DeleteReader_When_DeletionFails_Then_DeleteFailedIsReturned()
 
             result.Should().Be(ReaderChangeResult.DeleteFailed);
         }
+
+        [Fact, IsUnit]
+        public async Task DeleteReader_When_ReaderExists_Then_DeleteSubscriptionIsCalled()
```

What about tests for deletion failures? Can they happen?

---

I'm sure there can be deletion failures. If the subscriber does not exist the API spits a 404 I guess. I'll try to write tests to check the behavior of the methods though.

---

When 404 is returned, is that translated to an exception on our side or we simply swallow the error and go as nothing happened?
The error handling for service bus manager is bit unclear to me in case of deletion errors.